### ISSUE TITLE
Feature request: Add ability to "exclude current product" from the Product Collection block

### ISSUE THREAD

**Issue by opr** [woothemes/woocommerce#42572]
## Is your feature request related to a problem? Please describe.
I have a store and I am using the "All Products Block" on the product page. I would like to display "other" products in my store with the exception of the one the user is currently on.

## Describe the solution you'd like
I think an option in the editor to exclude the current product would be great. The option could even be hidden if the current post is not of type `product`.

## Describe alternatives you've considered
I've tried Hand Picked Products, but in a store with continually changing inventory it would be inconvenient to edit every product's instance of the Hand Picked Products block. All products achieves the desired outcome, but it contains the current product too.

## Additional context
To enable the block editor for a single product I used:

```php
function kogarashi_activate_gutenberg_for_product( $can_edit, $post_type )
{
		if ( $post_type == 'product' ) {
			$can_edit = true;
		}
	return $can_edit;
}

add_filter(
	'use_block_editor_for_post_type',
	'kogarashi_activate_gutenberg_for_product',
	20,
	2
);
```

**Comment by sunyatasattva** [comment]
Hey @opr excellent issue and use-case. We are going towards deprecating the All Products block very soon, in favour of the new one supported by the Product Query block.

I wonder if we should keep this issue and re-label it to the Product Query block as to accumulate ideas on certain implementation details. What do you think?

**Comment by opr** [comment]
Sounds good @sunyatasattva  :)

**Comment by manospsyx** [comment]
@xristos3490 @imanish003 @kmanijak the way we approached this in Product Recommendations, if I remember correctly, was to **automatically hide some products depending on the current context** --

- Single Product: Hide the current product.
- Cart/Checkout: Hide the cart items.
- Order: Hide the order items. 

No critical feedback was ever received about this -- seems to have done the job.

Just another good place to utilize Context (both in the Editor and in the storefront).

**Comment by kmanijak** [comment]
@xristos3490 is that something you're going to address on a query level? I'm asking if that's ok we assign it to SomewhereWarm?

**Comment by xristos3490** [comment]
Yes, this should be a patch that follows up on the context work! Feel free to assign me here, @kmanijak!

**Comment by gigitux** [comment]
Hey folks, which is the status of this issue?

**Comment by kmanijak** [comment]
It wasn't addressed "globally" to my best knowledge. What I mean is Upsells and Cross-Sells have that logic included but not the other collections.

And I don't think there's a plan to address it anytime soon as we're not focusing on Product Collection.

@xristos3490, do you know if that was addressed as part of Product Recommendations? Also, are you ok with closing this issue or you want to have it in SomewhereWarm scope?

**Comment by kmanijak** [comment]
After some thoughts, I don't think we'll be addressing it anytime soon considering current priorities. Hence I'm closing the issue as "won't do". It can be reopen later if needed.

**Comment by xristos3490** [comment]
> do you know if that was addressed as part of Product Recommendations?

It's not handled directly in the query, though the nature of the Product Recommendation collections won't include the current source product. So, it's a noop for this integration. That said, SWW is okay with marking this as won't fix for now.
